File: solutions/python/hamming/1/hamming.py
def distance(strand_a, strand_b):
    strands_len = len(strand_a)
    if strands_len != len(strand_b):
        raise ValueError("Strands must be of equal length.")
    hamming_distance = 0
    for i in range(0, strands_len):
        if strand_a[i] != strand_b[i]:
            hamming_distance += 1
    return hamming_distance

File: solutions/javascript/hamming/1/hamming.js
//
// This is only a SKELETON file for the 'Hamming' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const compute = (strand1, strand2) => {
  const len = strand1.length;
  if(len != strand2.length)
    throw new Error('strands must be of equal length');
  let hammingDistance = 0;
  for(let i = 0; i < len; i++) {
    if(strand1[i] != strand2[i])
      hammingDistance++;
  }
  return hammingDistance;
};
